frame/election-provider-multi-phase/src/lib.rs

```diff
@@ -738,6 +738,28 @@ pub mod pallet {
 	#[pallet::hooks]
 	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
 		fn on_initialize(now: T::BlockNumber) -> Weight {
+			if Self::force_phase().is_some() {
+				match Self::force_phase() {
+					Some(Phase::Signed) => {
+						// If the current phase is not `Phase::Off` proceed to the next round.
+						if !Self::current_phase().is_off() {
+							Self::do_force_rotate_round();
+						}
+
+						match Self::create_snapshot() {
```

Comment:
Could we reuse code either by 
- extracting in a separate `fn start_signed()`
- adding the check for force phase inside the existing arm
```
						match current_phase {
				Phase::Off if force_phase == Phase::Signed || remaining <= signed_deadline && remaining > unsigned_deadline => {
```

Reply:
A good reason to extract logic into functions `start_<some>_phase` is that we can potentially add it to trait `ElectionProvider` such that other pallets can tell EPM to force phase as well.